core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,7 +285,7 @@ private[spark] object Utils extends Logging {
         } else {
           // Restrict file permissions via chmod if available.
           // For Windows this step is ignored.
-          if (!isWindows && !chmod700(dir)) {
+          if (!isWindows && (conf==null || !isRunningInYarnContainer(conf)) && !chmod700(dir)) {
```

Comment:
I'd open a new PR vs master. The logic here is getting a little hairy, although it's not crazy. How about just setting permission to 750?